concepts/enums/introduction.md

```diff
@@ -0,0 +1,87 @@
+# Introduction
+
+## Enumerations
+
+So far we have mostly handled numbers and strings.
+To model the real world, we might want a limited number of values that a variable can take on.
+You might want a dedicated type with a few distinct values with distinct names.
+In a skateboard factory, the deck material can be selected from maple, bamboo, or plastic.
+
+You could use integers to encode those values, but would have to use extra code to check if there is an invalid value coming from the system for the material.
+The meaning of those [_magic numbers_][magic numbers] is difficult to trace over the source code and they are prone to mix-ups.
+This is a case for `enumerations`.
+
+## Example usage
+
+The specific term for this kind of enumeration is `scoped enumeration`.
+The snippet below shows how to write a `DeckMaterial` `enumeration`.
+Note the `enum class` keyword and the `;` at the end of the definition:
+
+```cpp
+enum class DeckMaterial {
+    maple,
+    bamboo,
+    plastic
+};
+```
+Now, look at a pricing function in the skate shop and take note of the scope resolution operator (`::`) to specify an `enumerator` from the `enumeration`:
+
+```cpp
+double deck_price(double base_price, DeckMaterial material) {
+    if(material == DeckMaterial::plastic) {
+        return base_price * 0.9;
+    }
+    return base_price * 1.3;
+}
+```
+
+## Scoping
+
+Imagine you have a second `enumeration` for the material of the wheels:
+
+```cpp
+enum class WheelMaterial {
+    steel,
+    clay,
+    plastic
+};
+```
+Although the wheels and the deck can both be made of _plastic_, the two cannot be confused.
+Each `enumeration` will have its `enumerators` in its own scope - its own `namespace`.
+This is the reason they are called `scoped enumerations`.
+
+~~~~exercism/advanced
+## Unscoped Enumerations
+
+You might think that there should be `unscoped` enumerations as well and you are correct.
+`Unscoped enumerations` lost favor to the version above, because they populate the global namespace.
```

Comment:
```suggestion
`Unscoped enumerations` have lost favor over time, because they can clutter up the global namespace.
```

Reply:
Another word instead of `clutter` could be `crowd`.  And you could add a qualification like:
`Unscoped enumerations` have lost favor over time, because they can  unnecessarily crowd the global namespace.

Clutter can also be qualified as:

 "`Unscoped enumerations` have lost favor over time, because they can create clutter in the global namespace."